File: packages/python/test/contract_smoke.py
"""Gate-2 contract smoke test (mirrors the Node contract.smoke.mjs).

Expects a Prism mock of the spec at MGX_MOCK_URL (default 127.0.0.1:4013).
Exercises each overlay namespace and asserts spec-shaped responses decode cleanly.

    node ../../MGX/tooling/sdk/contract/serve-mock.mjs --port 4013 &
    MGX_MOCK_URL=http://127.0.0.1:4013 python test/contract_smoke.py
"""
import os
import sys

from mgx.overlay import MgxClient, MgxApiError

base_url = os.environ.get("MGX_MOCK_URL", "http://127.0.0.1:4013")
mgx = MgxClient(base_url=base_url, access_token="mock")

failures = 0


def check(name, fn):
    global failures
    try:
        fn()
        print(f"  ok  {name}")
    except Exception as e:
        failures += 1
        detail = f"{e.status} {e.code}" if isinstance(e, MgxApiError) else str(e)
        print(f"  FAIL {name}: {detail}")


def _inventory_list():
    count = 0
    for lot in mgx.inventory.list(commodity="wheat"):
        assert isinstance(lot.id, str)
        count += 1
        if count >= 1:
            break


def _inventory_get():
    lot = mgx.inventory.get("inv_3Kd9aZ")
    assert lot is not None


def _inventory_filters():
    mgx.inventory.filters()


def _market_commodities():
    c = mgx.market.commodities()
    assert isinstance(c, list)


def _market_prices():
    p = mgx.market.prices(commodity="wheat")
    assert isinstance(p, list)


def _market_history():
    mgx.market.history("wheat", interval="day")


def _teams_list():
    t = mgx.teams.list()
    assert isinstance(t, list)


check("inventory.list yields items", _inventory_list)
check("inventory.get returns a listing", _inventory_get)
check("inventory.filters returns facets", _inventory_filters)
check("market.commodities returns array", _market_commodities)
check("market.prices returns array", _market_prices)
check("market.history returns series", _market_history)
check("teams.list returns array", _teams_list)

print("\nAll contract checks passed." if failures == 0 else f"\n{failures} check(s) failed.")
sys.exit(0 if failures == 0 else 1)

File: packages/node/test/contract.smoke.mjs
/**
 * Gate-2 contract smoke test. Boots nothing itself — expects a Prism mock of the
 * spec at MGX_MOCK_URL (default http://127.0.0.1:4010). Exercises each overlay
 * namespace against the mock and asserts spec-shaped responses decode cleanly.
 *
 *   node ../../../MGX/tooling/sdk/contract/serve-mock.mjs &   # in another shell
 *   npm run build && node test/contract.smoke.mjs
 */
import assert from 'node:assert'
import { MgxClient, MgxApiError } from '../dist/overlay/index.js'

const baseUrl = process.env.MGX_MOCK_URL ?? 'http://127.0.0.1:4010'
const mgx = new MgxClient({ baseUrl, accessToken: 'mock-token' })

let failures = 0
async function check(name, fn) {
  try {
    await fn()
    console.log(`  ok  ${name}`)
  } catch (e) {
    failures++
    console.error(`  FAIL ${name}: ${e instanceof MgxApiError ? `${e.status} ${e.code}` : e.message}`)
  }
}

await check('inventory.list yields items', async () => {
  let count = 0
  for await (const lot of mgx.inventory.list({ commodity: 'wheat' })) {
    assert.ok(typeof lot.id === 'string')
    if (++count >= 1) break
  }
})
await check('inventory.get returns a listing', async () => {
  const lot = await mgx.inventory.get('inv_3Kd9aZ')
  assert.ok(lot)
})
await check('inventory.filters returns facets', async () => {
  await mgx.inventory.filters()
})
await check('market.commodities returns array', async () => {
  const c = await mgx.market.commodities()
  assert.ok(Array.isArray(c))
})
await check('market.prices returns array', async () => {
  const p = await mgx.market.prices({ commodity: 'wheat' })
  assert.ok(Array.isArray(p))
})
await check('market.history returns series', async () => {
  await mgx.market.history('wheat', { interval: 'day' })
})
await check('teams.list returns array', async () => {
  const t = await mgx.teams.list()
  assert.ok(Array.isArray(t))
})
await check('bids.counter posts a counter', async () => {
  await mgx.bids.counter('bid_7Qp2', {
    quantityMt: 50,
    price: { amount: 300 },
    delivery: { from: new Date('2026-08-01'), to: new Date('2026-09-30') },
  })
})
await check('cashBids.offers yields offers', async () => {
  let n = 0
  for await (const o of mgx.cashBids.offers('cb_9Xa1')) {
    assert.ok(typeof o.id === 'string')
    if (++n >= 1) break
  }
})
await check('webhooks.list returns array', async () => {
  const w = await mgx.webhooks.list()
  assert.ok(Array.isArray(w))
})
await check('webhooks.create returns subscription', async () => {
  const w = await mgx.webhooks.create({ url: 'https://example.com/hook', events: ['trade.created'] })
  assert.ok(w)
})

console.log(failures === 0 ? '\nAll contract checks passed.' : `\n${failures} check(s) failed.`)
process.exit(failures === 0 ? 0 : 1)
